bootstrap_test.py

```diff
@@ -280,6 +280,33 @@ def test_consistent_range_movement_true_with_rf1_should_fail(self):
     def test_consistent_range_movement_false_with_rf1_should_succeed(self):
         self._bootstrap_test_with_replica_down(False, rf=1)
 
+    def test_rf_gt_nodes_multidc_should_succeed(self):
+        cluster = self.cluster
+        cluster.set_environment_variable('CASSANDRA_TOKEN_PREGENERATION_DISABLED', 'True')
+        cluster.populate([1, 1])
+        cluster.start()
+
+        node1 = cluster.nodelist()[0]
+        session = self.patient_cql_connection(node1)
+        session.execute("CREATE KEYSPACE k WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'dc1' : '2'}")
+        session.execute("CREATE TABLE k.testgtrfmultidc (KEY text PRIMARY KEY)")
+        session.execute("INSERT INTO k.testgtrfmultidc (KEY) VALUES ('test_rf_gt_nodes_multidc_should_succeed')")
```

Comment:
We could easily verify here that the log warnings are emitted in both nodes, since we don't check that in unit tests:
```python
warning = 'Your replication factor 2 for keyspace k is higher than the number of nodes 1 for datacenter dc1'
if cluster.version() >= '4.0':
    for node in cluster.nodelist():
        assert len(node.grep_log(warning)) == 3
```
Then we could verify that the warning is not emitted for the third node, since we have reached cluster size that matched the RF:
```python
if cluster.version() >= '4.0':
    assert len(node3.grep_log(warning)) == 0
```

Reply:
+1